clientTest/clientTest.go

```diff
@@ -42,56 +48,105 @@ func RunClientSideTest(res *http.Response, client *http.Client, wg *sync.WaitGro
-
+				
 				for _, attribute := range token.Attr {
 					if urlAttributes[attribute.Key] {
+						
 						clientSideTime := time.Now()
-						var assetUrl string
-
-						//maybe better logic for this is possible?
-						if strings.HasPrefix(attribute.Val, REQ_ADDRESS) {
-							assetUrl = attribute.Val
-						} else if strings.HasPrefix(attribute.Val, "http") {
-							assetUrl = attribute.Val
-						} else {
-							assetUrl = REQ_ADDRESS + attribute.Val
+						var assetUrl = AssetURL(attribute.Val, REQ_ADDRESS)
+						downloadedElement, err := client.Get(assetUrl)
+
+						if err != nil {
+							fmt.Println(err)
 						}
 
-						res, err := client.Get(assetUrl)
+						downloadedElement.Body.Close()
+						var logMessage = "finished downloading a "+ token.Data+ " file: "+ attribute.Val+ ", it took "
+						totalClientTime = LogElapstedTime(clientSideTime, totalClientTime, logMessage, DETAILED_LOGGING)
+
+						break
+					}
+				}
+
+			}else if token.Data == jsTag {
+
+				for _, attribute := range token.Attr {
+
+					//case where javascript needs to be downloaded first
+					if urlAttributes[attribute.Key] {
+
+						clientSideTimeDownLoad := time.Now()
+						var assetUrl = AssetURL(attribute.Val, REQ_ADDRESS)
+						script, err := client.Get(assetUrl)
 
 						if err != nil {
 							fmt.Println(err)
 						}
 
-						res.Body.Close()
+						var logMessageDownload = "downloaded some JS! from " + attribute.Val + " it took "
+
+						totalClientTime = LogElapstedTime(clientSideTimeDownLoad, totalClientTime, logMessageDownload, DETAILED_LOGGING)
 
-						endClientSideTime := time.Now().Sub(clientSideTime)
+						//now execute the javascript and record time to execute + time to download
+						if script != nil {
 
-						totalClientTime = totalClientTime + endClientSideTime.Seconds()
+							totalClientTimeJS := time.Now()
+							jsvm.Run(script)
+							var logMessageJS = "ran some JS! it took "
 
-						if DETAILED_LOGGING {
-							fmt.Println("finished downloading a ", token.Data, " file: ", attribute.Val, ", it took ", endClientSideTime)
+							totalClientTime = LogElapstedTime(totalClientTimeJS, totalClientTime, logMessageJS, DETAILED_LOGGING)
+
+							
+						}else{
+							fmt.Println("ERROR, No js code to execute")
 						}
 
-						break
+						script.Body.Close()
 					}
+
 				}
 
 			}
+			
 		}
 
 	}
 
-	return totalClientTime
+}
+
+func AssetURL(url string, REQ_ADDRESS string) string {
 
+	var assetUrl string
+	//maybe better logic for this is possible?
+	if strings.HasPrefix(url, REQ_ADDRESS) {
+		assetUrl = url
+	} else if strings.HasPrefix(url, "http") {
+		assetUrl = url
+	} else {
+		assetUrl = REQ_ADDRESS + url
+	}
+
+	return assetUrl
 }
+
+func LogElapstedTime(clientSideTime time.Time, totalClientTime float64, logMessage string, DETAILED_LOGGING bool) float64{
+
+	endClientSideTime := time.Now().Sub(clientSideTime)
+	totalClientTime = totalClientTime + endClientSideTime.Seconds()
+
+	if DETAILED_LOGGING {
```

Comment:
clientTest.go needs to be moved to the same directory ad loadhistogram.go. The package changed to main. the "clientTest." prefix to the methods removed and the variable can be shared between the files. 

I can do this after you merge this so it doesnt mess up all the work?